server/src/controllers/beneficiaries.controllers.ts

```diff
@@ -0,0 +1,43 @@
+import { Request, Response } from 'express';
+import BeneficiaryModel from '../models/BeneficiaryModel'; // Assuming Beneficiary is a Mongoose model
+import { auth } from '../../config/firebase';
+
+const editBeneficiary = async (req: Request, res: Response): Promise<void> => {
+    const beneficiaryId = req.query?._id;
+
+    try {
+        console.log(req.body);
+        if (beneficiaryId) {
+            const beneficiary = await BeneficiaryModel.findByIdAndUpdate(
+                beneficiaryId,
+                req.body,
+                { new: true } 
+            );
+
+            console.log(beneficiary);
+            res.status(200).json(beneficiary);
+        } else {
+            res.status(400).send({ message: "Missing Beneficiary ID" });
+        }
+    } catch (error: any) {
```

Comment:
Please handle errors in typesafe way. See example here for how to do this: 
https://github.com/ChangePlusPlusVandy/vac-net/blob/a9c4e9bab9f8ff75490e1b4e561bcb6290a39fbc/server/src/controllers/session.controllors.ts#L19